### DISCUSSION TITLE
Custom collectors with python.d plugin do not work

### DISCUSSION THREAD

**Discussion by ahdayusof** [netdata/netdata#14463] [Q&A]
Hi, I use a custom collector with python.d plugin for sensors where it will record all the sensors listed in the gateway. I have been working with Netdata 1.37.x and it works fine. But when I upgrade to Netdata version 1.38.0, the system does not work and no longer asks the server to get the data value.

**Comment by ilyam8** [comment]
Hi, @ahdabelle. And what is the error? Did you try to run the collector in debug mode?

**Reply by ahdayusof** [reply]
![image](https://user-images.githubusercontent.com/83644296/217504292-5691f3a8-9dce-4cf9-a150-383099b6f6b3.png)

I have the same error in Netdata v1.37.x however it works and I can see the charts, but in the new version I didn't see these charts in the netdata dashboard.

**Reply by ilyam8** [reply]
> error on loading source: PermissionError

It says it can't load the source code. Check the file ownership/permissions (compare with other files in the same directory).

**Reply by ahdayusof** [reply]
It has the same "root" owner as the other files in the same directory.

**Reply by ilyam8** [reply]
Ok, then check permissions. Netdata runs as `netdata` user. We didn't change anything in python.d.plugin in v1.38.0.

**Reply by ahdayusof** [reply]
I use this command to install netdata as root :

`wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh`

It seems to have downloaded all the files whose owner is root. And there is no netdata user. Where did I go wrong?

**Reply by ilyam8** [reply]
it is ok to be owned by root. As I said - check the file permissions (-rw-r--r-- thing).

**Reply by ahdayusof** [reply]
I tried to change the file permissions to 777 but it still doesn't work. But when I tried to change the permissions of the netdata directory, it works now. 

Do you have any tips for downloading netdata with netdata as the owner of the directory? What should I do before running the command to install netdata?

**Reply by ilyam8** [reply]
`root:root` is correct. User `netdata` (and any other user) can read them because those files are world-readable (-rw-r--**r--**). I don't know why you get the permission error.

> But when I tried to change the permissions of the netdata directory, it works now.

What directory? Can you show the full path? And ownership/permissions before and after your change.

**Reply by ahdayusof** [reply]
![image](https://user-images.githubusercontent.com/83644296/217746746-404e4c6a-94fe-4a52-b527-32e1f0bef8c8.png)

![image](https://user-images.githubusercontent.com/83644296/217746871-3358295a-44bb-44fb-8941-ce91667b1f30.png)

These are before I changed the netdata directories permissions to 777. 

I made some changes to these files, that's why I changed the permissions on the netdata directories and these files permissions :
•	**/usr/libexec/netdata**/python.d/sensors.chart.py
•	**/usr/lib/netdata**/conf.d/python.d/sensors.conf 
•	**/usr/lib/netdata**/conf.d/python.d/anomalies.conf 

However, I tried uninstalling netdata and reinstalling it by creating a netdata user (adduser) from the command line to see if the problem still occurs when I modify the files, and it still doesn't seem to work. 
And I tried again to change the file permissions after re-installing netdata, and it doesn't work anymore.

**Reply by ahdayusof** [reply]
It seems like python.d plugin is not running? 

![image](https://user-images.githubusercontent.com/83644296/217796651-91423742-bc9e-4761-bec2-68420b8ef29d.png)

I tried to check "python.d" in the error.log, it gave me this

![image](https://user-images.githubusercontent.com/83644296/217797469-3c12472e-a7a7-4ee0-b78f-62c9b40185c7.png)

**Reply by ilyam8** [reply]
check `collector.log` too

**Reply by ahdayusof** [reply]
I did `grep sensors /var/log/netdata/collector.log`

![image](https://user-images.githubusercontent.com/83644296/217803660-5409b37c-b522-44a6-86f8-0b6c82cfc1ca.png)

**Reply by ilyam8** [reply]
I don't follow all the changes you did. Do you get PermissionError if you try to run other python collectors? If you think there is a bug and you have problems after installing Netdata I suggest you open a bug report and describe steps to reproduce, your OS, etc.

**Reply by ahdayusof** [reply]
No, it seems that the PermissionsError was due to my code in the chart.py. I modified my sensor file. Now there is no more error, but the charts are still not visible in the dashboard. 

![image](https://user-images.githubusercontent.com/83644296/217806302-033517da-bb4e-4907-93b7-71fa0ae5b9c2.png)
